CanonicalLayouts/list-detail-compose/app/src/main/java/com/example/listdetailcompose/ui/ListDetailSample.kt

```diff
@@ -73,70 +70,40 @@ private data class DefinedWord(
     val definition: String
 )
 
+@OptIn(ExperimentalMaterial3AdaptiveApi::class)
 @Composable
-fun ListDetailSample(
-    windowSizeClass: WindowSizeClass,
-    displayFeatures: List<DisplayFeature>
-) {
-    // Query for the current window size class
-    val widthSizeClass by rememberUpdatedState(windowSizeClass.widthSizeClass)
-
-    /**
-     * The index of the currently selected word, or `null` if none is selected
-     */
+fun ListDetailSample() {
     var selectedWordIndex: Int? by rememberSaveable { mutableStateOf(null) }
+    val navigator = rememberListDetailPaneScaffoldNavigator<Nothing>()
 
-    /**
-     * True if the detail is currently open. This is the primary control for "navigation".
-     */
-    var isDetailOpen by rememberSaveable { mutableStateOf(false) }
+    BackHandler(enabled = navigator.canNavigateBack()) {
+        navigator.navigateBack()
+    }
 
-    ListDetail(
-        isDetailOpen = isDetailOpen,
-        setIsDetailOpen = { isDetailOpen = it },
-        showListAndDetail =
-            when (widthSizeClass) {
-                WindowWidthSizeClass.Compact, WindowWidthSizeClass.Medium -> false
-                WindowWidthSizeClass.Expanded -> true
-                else -> true
-            },
-        detailKey = selectedWordIndex,
-        list = { isDetailVisible ->
+    ListDetailPaneScaffold(
+        directive = navigator.scaffoldDirective,
+        value = navigator.scaffoldValue,
+        listPane = {
             val currentSelectedWordIndex = selectedWordIndex
             ListContent(
                 words = sampleWords.map(DefinedWord::word),
-                selectionState = if (isDetailVisible && currentSelectedWordIndex != null) {
```

Comment:
I think removing `isDetailVisible` here will result in showing a "selection" even when just the list is visible, which isn't ideal. I think we can read from the `navigator.scaffoldValue` to derive the equivalent logic of `isDetailVisible`:

```
navigator.scaffoldValue[ListDetailPaneScaffoldRole.Detail] == PaneAdaptedValue.Expanded
```

Reply:
Fixed in the following commit